Preparing all environments during prebuild

shaal opened the pull request:
# The Problem/Issue/Bug
Fixes #44 

## How this PR Solves The Problem
Create ready-made environments for all Drupal supported versions, in 3 site-installs (minimal, standard, demo_umami)

Extension should allow choosing between:
* QuickStart (fast)
* Full re-install (slow)

When QuickStart is selected, the ready-made environment (`/ready-made-envs`) should be copied to the main directory
Afterwards, patches can be applied and other modules can be downloaded.

# Additional details
I think I figured a trick to reduce the prebuild time, from 25 minutes to under 2 minutes:
The goal is to minimize DrupalPod's downtime when a prebuild has to run.

The idea is that before merging a PR into DrupalPod, I'll enter the workspace of that PR, and run the script that prepare the environments.
Running that script takes about 25 minutes, because it runs 3 profile installations (minimal, standard, umami) for each of the 5 Drupal supported versions.
`~8.9`
`8.9.x-dev`
`~9.2'`
`9.2.x-dev`
`9.3.x-dev`
At the end of the script, it creates tar.gz file from all the directories and files generated, including the `ddev snapshot` backups. (2.8GB -> 190MB)
Then I run `npx serve` and configure the port it's using (5000) public.
I will merge PR to main and add a commit of the URL where all the directories can be fetched.

The result - instead of waiting for a script to run in prebuild for 25 minutes, the prebuild only needs to download tar.gz file and extract it, this is done across 2 gitpod workspaces and takes less than 1 minute.
:trophy:

## Manual Testing Instructions

## Related Issue Link(s)

## Release/Deployment notes
<!-- Does this affect anything else, or are there ramifications for other code? Does anything have to be done on deployment? -->


<a href="https://gitpod.io/#https://github.com/shaal/DrupalPod/pull/46"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

Decipher commented:
@shaal

The first test I did, 9.2, no profile:
1. Failed the first time, but pretty certain it was a random failure.
2. Finished the second time, but had installed the Umami profile, I considered maybe this was a user error thing,
3. Finished again, definitely set 9.2 no profile, it still had a Umami profile install

shaal commented:
Thank you @Decipher, I found the cause for this edge-case and fixed it.

Link to replicate (now fixed) the issue - 
https://gitpod.io/#DP_PROJECT_NAME=druxt,DP_ISSUE_FORK=druxt-3249293,DP_ISSUE_BRANCH=feature%2F3249293-missing_resource_error,DP_PROJECT_TYPE=project_module,DP_MODULE_VERSION=1.1.x,DP_CORE_VERSION=9.2.0,DP_PATCH_FILE=,DP_INSTALL_PROFILE=''/https://github.com/shaal/DrupalPod/pull/46
